src/olympia/blocklist/mlbf.py

```diff
@@ -148,16 +166,18 @@ def soft_blocked_items(self) -> List[str]:
     def not_blocked_items(self) -> List[str]:
         all_blocks_ids = [version.version_id for version in self._all_blocks]
         not_blocked_items = MLBF.hash_filter_inputs(
-            Version.unfiltered.exclude(id__in=all_blocks_ids or ()).values_list(
-                'addon__addonguid__guid', 'version'
-            )
+            Version.unfiltered.exclude(id__in=all_blocks_ids or ())
+            .order_by('id')
+            .values_list('addon__addonguid__guid', 'version')
         )
         # even though we exclude all the version ids in the query there's an
         # edge case where the version string occurs twice for an addon so we
         # ensure not_blocked_items contain no blocked_items or soft_blocked_items.
-        return list(
-            set(not_blocked_items) - set(self.blocked_items + self.soft_blocked_items)
-        )
+        return [
+            item
+            for item in not_blocked_items
+            if item not in self.blocked_items + self.soft_blocked_items
```

Comment:
It looks like the second branch in this condition is not covered by tests.

With this diff applied, I would expect some tests to break but `test_mlbf.py` passes.
```diff
diff --git a/src/olympia/blocklist/mlbf.py b/src/olympia/blocklist/mlbf.py
index b5e7bbbfad..ed8d658452 100644
--- a/src/olympia/blocklist/mlbf.py
+++ b/src/olympia/blocklist/mlbf.py
@@ -176,7 +176,7 @@ class MLBFDataBaseLoader(BaseMLBFLoader):
         return [
             item
             for item in not_blocked_items
-            if item not in self.blocked_items + self.soft_blocked_items
+            if item not in self.blocked_items # + self.soft_blocked_items
         ]
```

Reply:
I tried reproducing this and both are covered by the duplicate test as we create a hard and soft block from the same reused addon